components/tidb_query/src/rpn_expr/impl_math.rs

```diff
@@ -11,6 +12,70 @@ pub fn pi() -> Result<Option<Real>> {
     Ok(Some(Real::from(std::f64::consts::PI)))
 }
 
+#[inline]
+#[rpn_fn]
+pub fn abs<A: Abs>(arg: &Option<A::Type>) -> Result<Option<A::Type>> {
+    if let Some(arg) = arg {
+        A::abs(arg)
+    } else {
+        Ok(None)
+    }
+}
+
+pub trait Abs {
+    type Type: Evaluable;
+
+    fn abs(arg: &Self::Type) -> Result<Option<Self::Type>>;
+}
+
+pub struct AbsInt;
+
+impl Abs for AbsInt {
+    type Type = Int;
+
+    #[inline]
+    fn abs(arg: &Self::Type) -> Result<Option<Self::Type>> {
+        match (*arg).checked_abs() {
+            None => Err(Error::overflow("BIGINT", &format!("abs({})", *arg)).into()),
```

Comment:
When there is an overflow, we try to handle it by an `EvalContext` first, if it can't be handled, then we report an error.

Reply:
Done, PTAL again :)